tests/Aspire.Hosting.Azure.Tests/AzureKeyVaultTests.cs

```diff
@@ -0,0 +1,164 @@
+              location: location
+              properties: {
+                tenantId: tenant().tenantId
+                sku: {
+                  family: 'A'
+                  name: 'standard'
+                }
+                enableRbacAuthorization: true
+              }
+              tags: {
+                'aspire-resource-name': 'mykv'
+              }
+            }
+
+            output vaultUri string = mykv.properties.vaultUri
+
+            output name string = mykv.name
+            """;
+        output.WriteLine(manifest.BicepText);
+        Assert.Equal(expectedBicep, manifest.BicepText);
+
+        var kvRoles = Assert.Single(model.Resources.OfType<AzureProvisioningResource>().Where(r => r.Name == $"mykv-roles"));
+        var kvRolesManifest = await AzureManifestUtils.GetManifestWithBicep(kvRoles, skipPreparer: true);
+        expectedBicep = """
+            @description('The location for the resource(s) to be deployed.')
+            param location string = resourceGroup().location
+
+            param mykv_outputs_name string
+
+            param principalType string
+
+            param principalId string
+
+            resource mykv 'Microsoft.KeyVault/vaults@2023-07-01' existing = {
+              name: mykv_outputs_name
+            }
+
+            resource mykv_KeyVaultSecretsUser 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+              name: guid(mykv.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '4633458b-17de-408a-b874-0445c86b69e6'))
+              properties: {
+                principalId: principalId
+                roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '4633458b-17de-408a-b874-0445c86b69e6')
+                principalType: principalType
+              }
+              scope: mykv
+            }
+            """;
+        output.WriteLine(kvRolesManifest.BicepText);
+        Assert.Equal(expectedBicep, kvRolesManifest.BicepText);
+    }
+
+    [Fact]
+    public async Task WithEnvironment_AddsKeyVaultSecretReference()
+    {
+        // Arrange: Create a test application builder.
+        using var builder = TestDistributedApplicationBuilder.Create();
+
+        // Add a key vault resource.
+        var kv = builder.AddAzureKeyVault("myKeyVault");
+
+        kv.Resource.SecretResolver = (s, ct) =>
+        {
+            return Task.FromResult<string?>("my secret value");
+        };
+
+        // Get a secret reference from the key vault resource.
+        var secretReference = kv.Resource.GetSecret("mySecret");
+
+        // Add a container resource that supports environment variables.
+        var containerBuilder = builder.AddContainer("myContainer", "nginx")
+                                       .WithEnvironment("MY_SECRET", secretReference);
+
+        var env = await containerBuilder.Resource.GetEnvironmentVariableValuesAsync(DistributedApplicationOperation.Run);
```

Comment:
Do we also want to verify the manifest? Or is that overkill?

Reply:
I can add it.